react-delegationdashboard/src/components/Trim/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import useDebounce from './useDebounce';
+
+interface TrimType {
+  text: string;
+  dataTestId?: string;
+}
+
+const Trim = ({ text, dataTestId = '' }: TrimType) => {
+  const [debounce, setDebounce] = React.useState(0);
+  const [overflow, setOverflow] = React.useState(false);
+  const wrapperRef = React.useRef(document.createElement('span'));
+  const childRef = React.useRef(document.createElement('span'));
+
+  const debounceTracker = useDebounce(debounce, 100);
+
+  const listener = () => {
```

Comment:
If this is used only for text overflow there's a way to achieve this only with css, ping me if that's the case and you need a hint :)